cartridge/failover/raft.lua

```diff
@@ -0,0 +1,137 @@
+local membership = require('membership')
+local checks = require('checks')
+local log = require('log')
+local topology = require('cartridge.topology')
+local pool = require('cartridge.pool')
+local argparse = require('cartridge.argparse')
+
+local vars = require('cartridge.vars').new('cartridge.failover')
+local PromoteLeaderError = require('errors').new_class('PromoteLeaderError')
+
+vars:new('leader_uuid')
+vars:new('raft_trigger')
+
+--- Generate appointments according to raft status.
+-- Used in 'raft' failover mode.
+-- @function get_appointments
+local function get_appointments(topology_cfg)
+    checks('table')
+    local replicasets = assert(topology_cfg.replicasets)
+
+    local appointments = {}
+
+    for replicaset_uuid, _ in pairs(replicasets) do
+        local leaders = topology.get_leaders_order(
+            topology_cfg, replicaset_uuid
+        )
+        if replicaset_uuid == vars.replicaset_uuid then
+            local my_leader_id = box.info.election.leader
```

Comment:
nit: box.info is not so cheap call. Probably it's better to call it once. (it's not serious because it's not hot path).